pkg/docker/docker_test.go

```diff
@@ -24,201 +30,336 @@ func TestContainerName(t *testing.T) {
+			t.Errorf("Test - %s: Expected error: Got: %+v", test, err)
 		}
 	}
 }
-
 func TestCheckAndPullImage(t *testing.T) {
+	// in addition to the deltas mentioned up top, kube fake client can't distinguish between exists and successful pull, and
+	// pull options with engine-api don't reference repo name like go-dockerclient did ... removed tests related to that
 	type testDef struct {
-		imageName           string
-		docker              test.FakeDockerClient
-		expectedImage       *docker.Image
-		expectedError       int
-		expectedPullOptions docker.PullImageOptions
+		imageName     string
+		docker        test.FakeDockerClient
+		calls         []string
+		expectedError string
 	}
-	image := &docker.Image{}
 	imageExistsTest := testDef{
 		imageName: "test_image",
-		docker: test.FakeDockerClient{
-			InspectImageErr:    []error{nil},
-			InspectImageResult: []*docker.Image{image},
-		},
-		expectedImage: image,
-	}
-	imagePulledTest := testDef{
-		imageName: "test_image",
-		docker: test.FakeDockerClient{
-			InspectImageErr:    []error{docker.ErrNoSuchImage, nil},
-			InspectImageResult: []*docker.Image{nil, image},
-		},
-		expectedImage:       image,
-		expectedPullOptions: docker.PullImageOptions{Repository: "test_image:" + DefaultTag},
+		docker:    test.FakeDockerClient{},
+		calls:     []string{"inspect_image"},
 	}
 	inspectErrorTest := testDef{
-		imageName: "test_image",
-		docker: test.FakeDockerClient{
-			InspectImageErr:    []error{docker.ErrConnectionRefused},
-			InspectImageResult: []*docker.Image{nil},
-		},
-		expectedImage: nil,
-		expectedError: errors.InspectImageError,
-	}
-	pullErrorTest := testDef{
-		imageName: "test_image",
-		docker: test.FakeDockerClient{
-			PullImageErr:       docker.ErrConnectionRefused,
-			InspectImageErr:    []error{nil},
-			InspectImageResult: []*docker.Image{nil},
-		},
-		expectedImage:       nil,
-		expectedError:       errors.PullImageError,
-		expectedPullOptions: docker.PullImageOptions{Repository: "test_image:" + DefaultTag},
-	}
-	errorAfterPullTest := testDef{
-		imageName: "test_image:testtag",
-		docker: test.FakeDockerClient{
-			InspectImageErr:    []error{docker.ErrNoSuchImage, docker.ErrNoSuchImage},
-			InspectImageResult: []*docker.Image{nil, image},
-		},
-		expectedImage:       nil,
-		expectedError:       errors.InspectImageError,
-		expectedPullOptions: docker.PullImageOptions{Repository: "test_image:testtag"},
+		imageName:     "test_image",
+		docker:        test.FakeDockerClient{},
+		calls:         []string{"inspect_image", "pull"},
+		expectedError: "unable to get test_image:latest",
 	}
 	tests := map[string]testDef{
-		"ImageExists":    imageExistsTest,
-		"ImagePulled":    imagePulledTest,
-		"InspectError":   inspectErrorTest,
-		"PullError":      pullErrorTest,
-		"ErrorAfterPull": errorAfterPullTest,
+		"ImageExists":  imageExistsTest,
+		"InspectError": inspectErrorTest,
```

Comment:
i'll respond to your summary comment about the fake docker stub about the options to consider...